indexing-hadoop/src/main/java/io/druid/indexer/path/StaticPathSpec.java

```diff
@@ -96,9 +114,33 @@ public final static void addToMultipleInputs(
     // MultipleInputs.addInputPath(job, path, inputFormatClassToUse)
     // but have to handle hadoop glob path ourselves correctly
     // This change and HadoopGlobPathSplitter.java can be removed once the hadoop issue is fixed
-    for (StringBuilder sb : HadoopGlobPathSplitter.splitGlob(path)) {
-      MultipleInputs.addInputPath(job, new Path(sb.toString()), inputFormatClassToUse);
+    Set<String> pathStrings = Sets.newLinkedHashSet();
+    for (String path : paths) {
+      Iterables.addAll(pathStrings, HadoopGlobPathSplitter.splitGlob(path));
+    }
+    if (!pathStrings.isEmpty()) {
+      addInputPath(job, pathStrings, inputFormatClassToUse);
+    }
+  }
+
+  // copied from MultipleInputs.addInputPath with slight modifications
+  private static void addInputPath(Job job, Iterable<String> pathStrings, Class<? extends InputFormat> inputFormatClass)
+  {
+    Configuration conf = job.getConfiguration();
+    StringBuilder inputFormats = new StringBuilder(Strings.nullToEmpty(conf.get(MultipleInputs.DIR_FORMATS)));
+
+    String[] paths = Iterables.toArray(pathStrings, String.class);
```

Comment:
I've a little revised in last patch and now path iterable returned from splitGlob are merged into a set first. So it seemed not need to them to be list.